File: src/main/java/com/godwin/jsonparser/util/NotificationUtil.java
package com.godwin.jsonparser.util;

import com.godwin.jsonparser.services.JsonPersistence;
import com.intellij.ide.BrowserUtil;
import com.intellij.notification.Notification;
import com.intellij.notification.NotificationAction;
import com.intellij.notification.NotificationType;
import com.intellij.notification.Notifications;
import com.intellij.openapi.actionSystem.AnActionEvent;
import org.jetbrains.annotations.NotNull;

import java.net.URI;

import static com.godwin.jsonparser.constants.ConstantsKt.TOOL_WINDOW_ID;

public class NotificationUtil {

    public static void showDonateNotification() {
        long now = System.currentTimeMillis();
        long daysBack = now - (10 * 24 * 60 * 60 * 1000); //10 days back
        long starClicked = JsonPersistence.Companion.getInstance().getStarClicked();
        long donateClicked = JsonPersistence.Companion.getInstance().getDonateClicked();
        if (starClicked == 1 || donateClicked == 1) {
            daysBack = now - (30L * 24 * 60 * 60 * 1000); //30 days back
        }
        long lastShownTime = JsonPersistence.Companion.getInstance().getJsonParserLastDisplayTime();
        if (lastShownTime < daysBack) {
            JsonPersistence.Companion.getInstance().setJsonParserLastDisplayTime(now);
            JsonPersistence.Companion.getInstance().setStarClicked(0);
            JsonPersistence.Companion.getInstance().setDonateClicked(0);
            Notification notification = new Notification(
                    TOOL_WINDOW_ID,
                    "Love it?",
                    "Love this plugin? Donate or Give it a star and spread the word",
                    NotificationType.INFORMATION);
            notification.addAction(new NotificationAction("Donate") {
                @Override
                public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
                    BrowserUtil.browse(URI.create("https://paypal.me/godwinj"));
                    JsonPersistence.Companion.getInstance().setDonateClicked(1);
                }
            });
            notification.addAction(new NotificationAction("Give it a star") {
                @Override
                public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
                    BrowserUtil.browse(URI.create("https://plugins.jetbrains.com/plugin/10650-json-parser"));
                    JsonPersistence.Companion.getInstance().setStarClicked(1);
                }
            });
            Notifications.Bus.notify(notification);
        }
    }
}

File: src/main/kotlin/com/godwin/jsonparser/util/NotificationUtil.kt
package com.godwin.jsonparser.util

import com.godwin.jsonparser.constants.TOOL_WINDOW_ID
import com.godwin.jsonparser.services.JsonPersistence
import com.godwin.jsonparser.util.analytics.AnalyticsConstant
import com.godwin.jsonparser.util.analytics.AnalyticsService
import com.intellij.ide.BrowserUtil
import com.intellij.notification.Notification
import com.intellij.notification.NotificationAction
import com.intellij.notification.NotificationType
import com.intellij.notification.Notifications
import com.intellij.openapi.actionSystem.AnActionEvent
import java.net.URI

object NotificationUtil {

    fun showDonateNotification() {
        val now = System.currentTimeMillis()
        val persistence = JsonPersistence.getInstance()
        val lastShownTime = persistence.jsonParserLastDisplayTime

        val daysBack = when {
            persistence.donateClicked == 1 -> now - (7L * 24 * 60 * 60 * 1000) // Weekly
            else -> now - (1L * 24 * 60 * 60 * 1000) // Daily

        }

        if (lastShownTime < daysBack) {
            persistence.apply {
                jsonParserLastDisplayTime = now
                starClicked = 0
                donateClicked = 0
            }

            val notification = Notification(
                TOOL_WINDOW_ID,
                "Love it?",
                "Love this plugin? Donate or Give it a star and spread the word",
                NotificationType.INFORMATION
            )

            notification.addAction(object : NotificationAction("Donate") {
                override fun actionPerformed(e: AnActionEvent, notification: Notification) {
                    AnalyticsService.track(AnalyticsConstant.NOTIFICATION_DONATE_CLICKED)
                    BrowserUtil.browse(URI.create("https://paypal.me/godwinj"))
                    persistence.donateClicked = 1
                }
            })

            notification.addAction(object : NotificationAction("Give it a star") {
                override fun actionPerformed(e: AnActionEvent, notification: Notification) {
                    AnalyticsService.track(AnalyticsConstant.NOTIFICATION_STAR_CLICKED)
                    BrowserUtil.browse(URI.create("https://plugins.jetbrains.com/plugin/10650-json-parser"))
                    persistence.starClicked = 1
                }
            })

            Notifications.Bus.notify(notification)
        }
    }

    fun showJsonRepairFailed() {
        AnalyticsService.track(AnalyticsConstant.REPAIR_FAILED)
        val notification = Notification(
            TOOL_WINDOW_ID,
            "This json can't be fixed.",
            "I coudn't fix your JSON file.",
            NotificationType.ERROR
        )
        Notifications.Bus.notify(notification)
    }

}
